plugins/system/webauthn/src/PluginTraits/AjaxHandlerLogin.php

```diff
@@ -118,153 +133,195 @@ public function onAjaxWebauthnLogin(): void
-			$credentialRepository,
-			$decoder,
-			$tokenBindingHandler,
-			$extensionOutputCheckerHandler,
-			$coseAlgorithmManager
-		);
+		// Is the user account blocked?
+		if ($user->block)
+		{
+			throw new RuntimeException(Text::_('JGLOBAL_AUTH_ACCESS_DENIED'));
+		}
 
-		// We init the Symfony Request object
-		$request = ServerRequestFactory::fromGlobals();
+		$statusSuccess = Authentication::STATUS_SUCCESS;
 
-		// Load the data
-		$publicKeyCredential = $publicKeyCredentialLoader->load($data);
-		$response            = $publicKeyCredential->getResponse();
+		$response                = $this->getAuthenticationResponseObject();
+		$response->status        = $statusSuccess;
+		$response->username      = $user->username;
+		$response->fullname      = $user->name;
+		// phpcs:ignore
+		$response->error_message = '';
+		$response->language      = $user->getParam('language');
+		$response->type          = 'Passwordless';
 
-		// Check if the response is an Authenticator Assertion Response
-		if (!$response instanceof AuthenticatorAssertionResponse)
+		if ($isAdmin)
 		{
-			throw new RuntimeException('Not an authenticator assertion response');
+			$response->language = $user->getParam('admin_language');
 		}
 
-		// Check the response against the attestation request
-		$userHandle = Joomla::getSessionVar('userHandle', null, 'plg_system_webauthn');
-		/** @var AuthenticatorAssertionResponse $authenticatorAssertionResponse */
-		$authenticatorAssertionResponse = $publicKeyCredential->getResponse();
-		$authenticatorAssertionResponseValidator->check(
-			$publicKeyCredential->getRawId(),
-			$authenticatorAssertionResponse,
-			$publicKeyCredentialRequestOptions,
-			$request,
-			$userHandle
-		);
+		/**
+		 * Set up the login options.
+		 *
+		 * The 'remember' element forces the use of the Remember Me feature when logging in with Webauthn, as the
+		 * users would expect.
+		 *
+		 * The 'action' element is actually required by plg_user_joomla. It is the core ACL action the logged in user
+		 * must be allowed for the login to succeed. Please note that front-end and back-end logins use a different
+		 * action. This allows us to provide the WebAuthn button on both front- and back-end and be sure that if a
+		 * used with no backend access tries to use it to log in Joomla! will just slap him with an error message about
+		 * insufficient privileges - the same thing that'd happen if you tried to use your front-end only username and
+		 * password in a back-end login form.
+		 */
+		$options = [
+			'remember' => true,
+			'action'   => 'core.login.site',
+		];
+
+		if ($isAdmin)
+		{
+			$options['action'] = 'core.login.admin';
+		}
+
+		// Run the user plugins. They CAN block login by returning boolean false and setting $response->error_message.
+		PluginHelper::importPlugin('user');
+		$event   = AbstractEvent::create('onUserLogin', [(array) $response, $options]);
```

Comment:
```suggestion
		$event   = new \Joomla\Event\Event('onUserLogin', [(array) $response, $options]);
```
No idea how this ever worked, using the AbstractEvent creator ends into an error log entry 'Received login failure. Message: No subject given for the onUserAfterLogin event'.

The `AbstractEvent::create` function needs a subject, it doesn't need any if you use the deprecated `tiggerEvent` function or directly create the event...

Reply:
This was a wrong search & replace. I'm fixing it now.